doc/api/cli.md

```diff
@@ -222,6 +222,57 @@ Both of the above may change in future updates, which will be breaking changes.
 added: v12.0.0
 -->
 
+### `--heap-prof`
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+Starts the V8 heap profiler on start up, and writes the heap profile to disk
+before exit.
+
+If `--heap-prof-dir` is not specified, the generated profile will be placed
+in the current working directory.
+
+If `--heap-prof-name` is not specified, the generated profile will be
+named `Heap.${yyyymmdd}.${hhmmss}.${pid}.${tid}.${seq}.heapprofile`.
+
+```console
+$ node --heap-prof index.js
+$ ls *.heapprofile
+Heap.20190409.202950.15293.0.001.heapprofile
+```
+
+### `--heap-prof-dir`
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+Specify the directory where the heap profiles generated by `--heap-prof` will
+be placed.
+
+### `--heap-prof-interval`
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+Specify the sampling interval in bytes for the heap profiles generated
```

Comment:
```suggestion
Specify the average sampling interval in bytes for the heap profiles generated
```

Sampling is done via a randomized poisson process with a mean that corresponds to the value provided here. Saying `average` here is important because it otherwise would imply that samples are taken at regular intervals; which is not the case.